runtime/compiler/net/CommunicationStream.hpp

```diff
@@ -160,6 +168,12 @@ class CommunicationStream
                {
                if (EINTR != errno)
                   {
+		  if(errno == _lastReadError)
+		     _readRetryAttempt++;
+		  else
+		     _readRetryAttempt = 0;
```

Comment:
`_readRetryAttempt` needs to be reset if we manage to read without errors.
Also, if the error is EAGAIN we are going to retry immediately, so we don't want to increment `_readRetryAttempt`

Reply:
For the` _readRetryAttempt` I will make the correction to reset the counter when the read is successful.
I did not understand the EAGAIN case. Currently for all `EINTR != errno` we throw StreamFailure exception hence will not try again.

Reply:
The exception has the last parameter "EAGAIN == errno" which sets `retryConnectionImmediately` in JITServerHelpers::postStreamFailure(OMRPortLibrary *portLibrary, TR::CompilationInfo *compInfo, bool retryConnectionImmediately, bool connectionFailure)

Reply:
Thanks for pointing that code, I missed noticing it. Even if we retry immediately, if we get same error which is thrown as StreamFailure exception then is it not good to try only M`AX_COMPILE_ATTEMPTS-1` times?

Reply:
OK.. No may be we have the special code for `EAGAIN` because we think its a temporary problem and we hope to have a successful read in the next attempt. So in which case as you suggested we should probably not increment `_readRetryAttempt`